packages/spacecat-shared-google-client/src/index.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { google } from 'googleapis';
+import { ok, internalServerError } from '@adobe/spacecat-shared-http-utils';
+import { OAuth2Client } from 'google-auth-library';
+
+export default class GoogleClient {
+  static async createFrom(context) {
+    const authClient = new OAuth2Client(
+      context.env.GOOGLE_CLIENT_ID,
+      context.env.GOOGLE_CLIENT_SECRET,
+      context.env.GOOGLE_REDIRECT_URI,
+    );
+
+    authClient.setCredentials({
+      access_token: context.env.ACCESS_TOKEN,
```

Comment:
are access token and refresh token expected to come from env variables? 🤔 these should be short lived creds, unless we don't have a way to automatically rotate them, I guess we need to obtain these tokens freshly from google, not from env variables

Reply:
the idea was to write a customer secret wrapper, that adds the secrets as env variables for the duration of the function execution.
Do you think this should be handled differently?

Reply:
> the idea was to write a customer secret wrapper, that adds the secrets as env variables for the duration of the function execution.

where the "customer secret wrapper" is expected to read those secrets from? request body or the query params? If that's the case, then we can get it from `context.data` or `context.params`